src/Glpi/Asset/AssetDefinition.php

```diff
@@ -604,6 +603,13 @@ private function getDecodedCapacitiesField(): array
             $this->fields['capacities'] = '[]'; // prevent warning to be triggered on each method call
             $capacities = [];
         }
+
+        if (!is_array($capacities)) {
+            // should not happen, do not trigger cleaning to prevent unexpected mass deletion of data
+            throw new \UnexpectedValueException(
+                sprintf('Invalid `capacities` value `%s`.', $this->fields['capacities'])
+            );
+        }
```

Comment:
How can this happen ? If `$capacities` is not an array, `validateCapacityArray()` will return false, and the variable will be forced to an empty array.

If it is an issue, you should replace the `trigger_error` by an exception.

Reply:
Indeed, this should not happen. Throwing exception on validate fail would be enough if needed. I'll take an eye.